### ISSUE TITLE
`neuron_parallel_compile` fails when an non utf-8 character

### ISSUE THREAD

**Issue by michaelbenayoun** [aws-neuron/aws-neuron-sdk#886]
`neuron_parallel_compile` relies on `torch_neuronx.parallel_compile.neuron_parallel_compile.get_hlos_from_run_log` to read the log and extract the names of the HLO graphs to compile.

 If the logs contain unexpected characters it fails and precompilation is only partial: the workers not failing compile their thing and the other are dead.

Suggested change:

```python
  def get_hlos_from_run_log(trial_run_log):
      # New graphs are detected by specific message matching key
      hlo_key = "Extracting graphs"
      new_hlo_list = []
      with codecs.open(trial_run_log, "r", encoding="utf-8", errors="ignore") as f:
          for line in f.readlines():
              # Move temporary MODULE_* files into workdir before checking if there are any
              # new graphs. In try_compilations, compile only new graphs (those without
              # corresponding neffs).
              if hlo_key in line:
                  model_path = line.split("Extracting graphs (")[1].split(")")[0]
                  new_hlo_list.append(model_path)

      format_str = '\n\t'
      LOGGER.info(f"New graph list from script: {format_str.join(new_hlo_list)}") 
      return new_hlo_list
   
```

**Comment by jeffhataws** [comment]
Thanks @michaelbenayoun . This is a good find. We will take a look at your suggestion.

**Comment by aws-rhsoln** [comment]
Hello, this should be fixed with the latest [2.19 release](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/general/setup/neuron-setup/pytorch/neuronx/ubuntu/torch-neuronx-ubuntu20.html#setup-torch-neuronx-ubuntu20) . Please give it a try and let us know if it works for you.
